### PR TITLE
[SPARK-11092] [Docs] Add source links to scaladoc generation

### PR DESCRIPTION
**Pull request by jodersky**
Modify the SBT build script to include GitHub source links for generated Scaladocs, on releases only (no snapshots).

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -531,8 +535,16 @@ object Unidoc {
...
     ),
 
-    // Group similar methods together based on the @group annotation.
-    scalacOptions in (ScalaUnidoc, unidoc) ++= Seq("-groups")
+    scalacOptions in (ScalaUnidoc, unidoc) ++= Seq(
+      "-groups" // Group similar methods together based on the @group annotation.
+    ) ++ (
+      // Add link to github sources when generating scaladoc for a non-snapshot release
+      if (!isSnapshot.value) {
+        Opts.doc.sourceUrl(s"https://github.com/apache/spark/tree/v${version.value}€{FILE_PATH}.scala")
```
Comment (srowen) [inline project/SparkBuild.scala]: I suspect this is fine if, y'know, it builds correctly and outputs the links you expect. Have you tried it?

The € is a typo, right?

**Comment by jodersky** [comment]
The € is not a typo, I don't know why it was chosen but it €{FILE_PATH} gets replaced by scaladoc with the corresponding source file.
Btw, I factored out the base URL to a separate setting in the newest commit

**Comment by jodersky** [comment]
It works fine on my machine, however maybe someone else should still test it. (You can just replace `isSnapshot` with `true` in case you don't want to change the version of your Spark fork)

**Comment by srowen** [comment]
OK thank you, I probably should have suspected it was on purpose. Seems OK to me.

**Pull request closed without merging**